src/view/xwayland.cpp

```diff
@@ -326,6 +327,25 @@ class wayfire_xwayland_view : public wayfire_xwayland_view_base
 
         on_request_move.set_callback([&] (void*) { move_request(); });
         on_request_resize.set_callback([&] (void*) { resize_request(); });
+        on_request_activate.set_callback([&] (void*)
+        {
+            if (!this->activated)
+            {
+                view_self_request_focus_signal data;
+                wf::output_t *view_output;
+                view_output = this->get_output();
+                data.view   = this;
+                if (view_output)
```

Comment:
I thought we discussed this a second time on IRC, and this signal should be emitted on the view and on core always?

Reply:
done